### PR TITLE
fluent-icon-theme: 2022-02-28 -> 2022-09-20

### PR DESCRIPTION
**Pull request by Icy-Thought**
###### Description of changes
Updating our `Fluent-Icon-Theme` to the latest available version.

>**Note**

`lib.checkListOfEnum` was merged into one line because both `nixpkgs-fmt` and `alejandra` were tripping on the multi-line version.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/fluent-icon-theme/default.nix [hunk 1]
```diff
@@ -1,73 +1,60 @@
...
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  gtk3,
+  hicolor-icon-theme,
+  jdupes,
+  roundedIcons ? false,
+  blackPanelIcons ? false,
+  colorVariants ? [],
+}: let
```
Comment (SuperSandro2000) [inline pkgs/data/icons/fluent-icon-theme/default.nix:11]:
```suggestion
{ lib
, stdenvNoCC
, fetchFromGitHub
, gtk3
, hicolor-icon-theme
, jdupes
, roundedIcons ? false
, blackPanelIcons ? false
, colorVariants ? []
}:

let
```
nixpkgs is not using alejandra and it is incompatible with nixpkgs-fmt and nix-fmt.

File: pkgs/data/icons/fluent-icon-theme/default.nix [hunk 2]
```diff
@@ -1,73 +1,60 @@
...
+    src = fetchFromGitHub {
+      owner = "vinceliuice";
+      repo = pname;
+      rev = version;
+      sha256 = "Ce8LTIxKabeqV9QVK68DqUVwtwG5lyxDPDQx0mLIr5o=";
+    };
+
+    nativeBuildInputs = [gtk3 jdupes];
+
+    buildInputs = [hicolor-icon-theme];
```
Comment (SuperSandro2000) [inline pkgs/data/icons/fluent-icon-theme/default.nix:28]:
```suggestion
    nativeBuildInputs = [ gtk3 jdupes ];

    buildInputs = [ hicolor-icon-theme ];
```

File: pkgs/data/icons/fluent-icon-theme/default.nix [hunk 3]
```diff
@@ -1,73 +1,60 @@
...
+
+      runHook postInstall
+    '';
+
+    meta = with lib; {
+      description = "Fluent icon theme for linux desktops";
+      homepage = "https://github.com/vinceliuice/Fluent-icon-theme";
+      license = licenses.gpl3Plus;
+      platforms = platforms.linux;
+      maintainers = with maintainers; [icy-thought];
```
Comment (SuperSandro2000) [inline pkgs/data/icons/fluent-icon-theme/default.nix:58]:
```suggestion
      maintainers = with maintainers; [ icy-thought ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/data/icons/fluent-icon-theme/default.nix (hunk 1 above)
Comment (Icy-Thought) [reply pkgs/data/icons/fluent-icon-theme/default.nix:11]: Was not aware of this fact. 
Will make sure to remember this when contributing in the future!

**Pull request merged**